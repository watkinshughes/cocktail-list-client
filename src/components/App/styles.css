.App h1 {
    font-size: 1.5em;
    font-weight: bold;
}

.App h2 {
    font-size: 1em;
    font-weight: bold;
}

.App a {
    display: block;
    width: 100%;
    padding: 0;
    border: 0;
    color: #000;
    text-align: left;
    text-decoration: none;
    cursor: pointer;
}

.App .display-linebreak {
    white-space: pre-line;
}