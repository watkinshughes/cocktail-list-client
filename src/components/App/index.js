import React, { Component, Fragment } from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Title from "../Title";
import CocktailList from "../CocktailList";
import AddCocktail from "../AddCocktail";
import CocktailDetails from "../CocktailDetails";
import styles from "./styles.module.css";

class App extends Component {
  render() {
    return (
      <Fragment>
        <Router>
          <Fragment>
            <header>
              <Link to="/">
                <Title copy="Cocktail List" />
              </Link>
              <Link className={styles.menu} to="/add-cocktail">
                Add cocktail
              </Link>
            </header>
            <Switch>
              <Route exact path="/" component={CocktailList} />
              <Route path="/add-cocktail" component={AddCocktail} />
              <Route path="/:id" component={CocktailDetails} />
            </Switch>
          </Fragment>
        </Router>
      </Fragment>
    );
  }
}

export default App;
