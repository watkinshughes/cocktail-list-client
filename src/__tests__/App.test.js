import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import App from "../components/App";

test("snapsot", () => {
  const c = create(<App />);
  expect(c.toJSON()).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
