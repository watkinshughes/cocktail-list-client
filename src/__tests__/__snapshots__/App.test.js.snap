// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapsot 1`] = `
Array [
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1
        className="title"
      >
        Cocktail List
      </h1>
    </a>
    <a
      className="menu"
      href="/add-cocktail"
      onClick={[Function]}
    >
      Add cocktail
    </a>
  </header>,
  <div
    className="loading"
  >
    <span
      className="visually-hidden"
    >
      Loading...
    </span>
  </div>,
]
`;
